--- libvorbis-libogg-android/jni/libvorbis-jni/Android.mk.orig	2014-06-17 19:22:50.621559073 +0200
+++ libvorbis-libogg-android/jni/libvorbis-jni/Android.mk	2014-06-17 19:38:20.641581140 +0200
@@ -4,9 +4,6 @@
 
 LOCAL_MODULE := vorbis-jni
 LOCAL_CFLAGS += -I$(LOCAL_PATH)/../include -fsigned-char
-ifeq ($(TARGET_ARCH),arm)
-	LOCAL_CFLAGS += -march=armv6 -marm -mfloat-abi=softfp -mfpu=vfp
-endif
 
 LOCAL_SHARED_LIBRARIES := libogg libvorbis
 
--- libvorbis-libogg-android/jni/libvorbis/Android.mk.orig	2014-06-17 19:22:39.077558797 +0200
+++ libvorbis-libogg-android/jni/libvorbis/Android.mk	2014-06-17 19:38:52.121581887 +0200
@@ -4,9 +4,6 @@
 
 LOCAL_MODULE := libvorbis
 LOCAL_CFLAGS += -I$(LOCAL_PATH)/../include -ffast-math -fsigned-char
-ifeq ($(TARGET_ARCH),arm)
-	LOCAL_CFLAGS += -march=armv6 -marm -mfloat-abi=softfp -mfpu=vfp
-endif
 LOCAL_SHARED_LIBRARIES := libogg
 
 LOCAL_SRC_FILES := \
--- libvorbis-libogg-android/jni/libogg/Android.mk.orig	2014-06-17 19:22:33.965558675 +0200
+++ libvorbis-libogg-android/jni/libogg/Android.mk	2014-06-17 19:38:25.337581252 +0200
@@ -4,10 +4,6 @@
 
 LOCAL_MODULE := libogg
 LOCAL_CFLAGS += -I$(LOCAL_PATH)/../include -ffast-math -fsigned-char
-ifeq ($(TARGET_ARCH),arm)
-	LOCAL_CFLAGS += -march=armv6 -marm -mfloat-abi=softfp -mfpu=vfp
-endif
-
 
 LOCAL_SRC_FILES := \
 	bitwise.c \
