--- irrlicht/source/Irrlicht/Android/CIrrDeviceAndroid.cpp.orig	2015-08-29 15:43:09.000000000 +0300
+++ irrlicht/source/Irrlicht/Android/CIrrDeviceAndroid.cpp	2016-05-13 21:36:22.880388505 +0300
@@ -486,7 +486,7 @@
 				event.KeyInput.Char = 0;
 			}
 
-			device->postEventFromUser(event);
+			status = device->postEventFromUser(event);
 		}
 		break;
 		default:
@@ -543,7 +543,7 @@
     KeyMap[1] = KEY_LBUTTON; // AKEYCODE_SOFT_LEFT
     KeyMap[2] = KEY_RBUTTON; // AKEYCODE_SOFT_RIGHT
     KeyMap[3] = KEY_HOME; // AKEYCODE_HOME
-    KeyMap[4] = KEY_BACK; // AKEYCODE_BACK
+    KeyMap[4] = KEY_CANCEL; // AKEYCODE_BACK
     KeyMap[5] = KEY_UNKNOWN; // AKEYCODE_CALL
     KeyMap[6] = KEY_UNKNOWN; // AKEYCODE_ENDCALL
     KeyMap[7] = KEY_KEY_0; // AKEYCODE_0
