--- a/libcharset/lib/localcharset.c	2015-06-10 11:55:25.933870724 +0200
+++ b/libcharset/lib/localcharset.c	2015-06-10 11:55:39.578063493 +0200
@@ -47,7 +47,7 @@
 
 #if !defined WIN32_NATIVE
 # include <unistd.h>
-# if HAVE_LANGINFO_CODESET
+# if HAVE_LANGINFO_CODESET && !(defined __ANDROID__)
 #  include <langinfo.h>
 # else
 #  if 0 /* see comment below */
@@ -124,7 +124,7 @@ get_charset_aliases (void)
   cp = charset_aliases;
   if (cp == NULL)
     {
-#if !(defined DARWIN7 || defined VMS || defined WIN32_NATIVE || defined __CYGWIN__)
+#if !(defined DARWIN7 || defined VMS || defined WIN32_NATIVE || defined __CYGWIN__ || defined __ANDROID__)
       const char *dir;
       const char *base = "charset.alias";
       char *file_name;
@@ -338,6 +338,9 @@ get_charset_aliases (void)
            "CP54936" "\0" "GB18030" "\0"
            "CP65001" "\0" "UTF-8" "\0";
 # endif
+# if defined __ANDROID__
+      cp = "*" "\0" "UTF-8" "\0";
+# endif
 #endif
 
       charset_aliases = cp;
@@ -361,7 +364,7 @@ locale_charset (void)
   const char *codeset;
   const char *aliases;
 
-#if !(defined WIN32_NATIVE || defined OS2)
+#if !(defined WIN32_NATIVE || defined OS2 || defined __ANDROID__)
 
 # if HAVE_LANGINFO_CODESET
 
