--- sqlite3.c	2016-11-29 02:29:24.000000000 +0000
+++ sqlite3.c	2016-12-08 22:54:54.206465377 +0000
@@ -30445,7 +30445,14 @@
 #if OS_VXWORKS
   struct vxworksFileId *pId;  /* Unique file ID for vxworks. */
 #else
-  ino_t ino;                  /* Inode number */
+  #ifdef ANDROID
+    // Bionic's struct stat has a 64 bit st_ino on both 32 and
+    // 64 bit architectures. ino_t remains 32 bits wide on 32 bit
+    // architectures and can lead to inode truncation.
+    unsigned long long ino;     /* Inode number */
+  #else
+    ino_t ino;     /* Inode number */
+  #endif
 #endif
 };
